packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -167,14 +166,14 @@ abstract class BoxBorder extends ShapeBorder {
   }
 
   @override
-  Path getInnerPath(Rect rect, { @required TextDirection textDirection }) {
+  Path getInnerPath(Rect rect, { TextDirection? textDirection }) {
     assert(textDirection != null, 'The textDirection argument to $runtimeType.getInnerPath must not be null.');
```

Comment:
Shouldn't this be `required TextDireciton textDirection`?

Reply:
The overriden method documents that textDirection can be null
https://github.com/flutter/flutter/blob/bdadd7f4e88b35c976bb3160a5c312fa4bbe09c4/packages/flutter/lib/src/painting/borders.dart#L470-L479

Reply:
Ahh ok. That makes sense.